lib/server/services/qualityControlFlag/QcFlagTypeService.js

```diff
@@ -95,6 +95,55 @@ class QcFlagTypeService {
         return qcFlagTypeAdapter.toEntity(await this.getById(newFlagInstance.id));
     }
 
+    /**
+     * Update existing QC Flag Type
+     * @param {number} [id] QC flag type id
+     * @param {Partial<QcFlagType>} [patch] QC flag type parameters
+     * @param {object} [relations] QC flag type entity relations
+     * @param {Partial<UserIdentifier>} [relations.user] user identifiers
+     * @return {Promise<QCFlagType>} promise
+     */
+    async update(id, patch, relations = {}) {
+        const {
+            name,
+            method,
+            bad,
+            color,
+            archived,
+        } = patch;
+        const { user: { userId, externalUserId } = {} } = relations;
+
+        const user = await getUserOrFail({ userId, externalUserId });
+
+        const qcFlagType = await QcFlagTypeRepository.findOne({ where: { id } });
+        if (!qcFlagType) {
+            throw new NotFoundError(`Quality Control Flag Type with this id (${id}) could not be found`);
+        }
+
+        const existingQcFlagType = await QcFlagTypeRepository.findOne({
+            where: { [Op.not]: { id }, [Op.or]: [name ? { name } : {}, method ? { method } : {}] },
```

Comment:
Here you may want to test only properties that are not equal to the existing QC flag type?

Reply:
no, here I check whether some other flag (not id) has same name or method

Reply:
Yes, but if someone want to update the QC flag method, and send the full qc flag, including the current name it should work?

Reply:
yes, `[Op.not]: { id } ` prevents target flag won't be fetched, I renamed the variable to `conflictingQcFlagType`

Reply:
Indeed